bundler/lib/bundler/ruby_dsl.rb

```diff
@@ -25,5 +20,26 @@ def ruby(*ruby_version)
       end
       @ruby_version = RubyVersion.new(ruby_version, options[:patchlevel], options[:engine], options[:engine_version])
     end
+
+    # Support the various file formats found in .ruby-version files.
+    #
+    #     3.2.2
+    #     ruby-3.2.2
```

Comment:
What about other ruby implementations, like jruby or truffleruby?

Reply:
This is a bit weird since specifying jruby or truffleruny require 3 values: version, engine, and engine version. I tested a bunch of ways to allow this. Bundler needs version, engine version and engine. .ruby-version doesn't support syntax like that.

The only solution I see is putting engine: and engine_version: in the options with file:, and having file be the underlying ruby version that is given by the engine. Bundler doesn't like getting the jruby's engine version as the ruby version, as far as I found in testing, so we'd have to load the ruby version from the environment if .ruby-version had jruby-0.9.2.0.

Reply:
With `asdf` only MRI rubies are the simple version.
Alternate values are the same as with `rbenv`, or `ruby-build` (which is what `asdf-ruby` uses under the hood).
Many of those are not numeric at all, or are numeric but the numbers are not based on MRI versions:
```
artichoke-dev
jruby-dev
jruby-1.7.27
jruby-9.4.3.0
maglev-2.0.0-dev
mruby-dev
mruby-3.2.0
picoruby-3.0.0
rbx-5.0
ree-1.8.7-2012.02
truffleruby-dev
truffleruby-23.0.0
truffleruby+graalvm-dev
truffleruby+graalvm-23.0.0
```